test/unittest/source/test_gateway.cpp

```diff
@@ -1629,16 +1629,21 @@ namespace casual
 
          std::map< std::string, int> domains;
 
-         algorithm::for_n< 9>( [ &domains]() mutable
+         algorithm::for_n< 100>( [ &domains]() mutable
          {
             auto buffer = local::call( "casual/example/domain/name");
             domains[ buffer.get()]++;
          });
 
-         // expect even distribution
-         EXPECT_TRUE( domains[ "B"] == 3);
-         EXPECT_TRUE( domains[ "C"] == 3);
-         EXPECT_TRUE( domains[ "D"] == 3);
+         // for a domain to have 0 calls after 100 events:
+         // ( 2 / 3 ) ^100 ~~ 1 in 4,065611775352153e17
+         // 1 in 400'000'000'000'000'000
+         // one in 400 million billions. 
```

Comment:
😄

Reply:
If we run the test non stop. On average it will fail every 1.2 billion years :)